src/main/java/com/hillel/items_exchange/controller/CategoryController.java

```diff
@@ -21,86 +24,63 @@
 
 @RestController
 @RequestMapping("/category")
-@CrossOrigin(origins = "http://localhost:4200")
 @RequiredArgsConstructor
-@Slf4j
 public class CategoryController {
 
     private final CategoryService categoryService;
 
     @GetMapping("/names")
-    public ResponseEntity<List<String>> getAllCategoriesNames() {
-        List<String> categoriesNames = categoryService.findAllCategoryNames();
-        if (categoriesNames.isEmpty()) {
-            return new ResponseEntity<>(HttpStatus.NOT_FOUND);
-        }
-
-        return new ResponseEntity<>(categoriesNames, HttpStatus.OK);
+    @ApiOperation(value = "Get all names of existing categories.")
+    public ResponseEntity<List<String>> getAllCategoryNames() {
+        return new ResponseEntity<>(categoryService.findAllCategoryNames(), HttpStatus.OK);
     }
 
     @GetMapping("/all")
+    @ApiOperation(value = "Get all existing categories.")
     public ResponseEntity<List<CategoryDto>> getAllCategories() {
-        List<CategoryDto> categories = categoryService.findAllCategories();
-        if (categories.isEmpty()) {
-            return new ResponseEntity<>(HttpStatus.NOT_FOUND);
-        }
-
-        return new ResponseEntity<>(categories, HttpStatus.OK);
+        return new ResponseEntity<>(categoryService.findAllCategoryDto(), HttpStatus.OK);
     }
 
     @GetMapping("/{category_id}")
+    @ApiOperation(value = "Get a category by its ID, if it exists")
     public ResponseEntity<CategoryDto> getCategoryById(@PathVariable("category_id") long id) {
-        return categoryService.findCategoryById(id)
+        return categoryService.findCategoryDtoById(id)
```

Comment:
You could make it even simpler and to return OK or NOT FOUND statuses doing that: ``return ResponseEntity.of(categoryService.findCategoryDtoById(id))`` because you work with Optional here.

Reply:
Great approach!